src/components/molecules/AutocompleteInput.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+<div class="autocomplete-input">
+  <input required autofocus
+    type="text"
+    placeholder="Search your school" 
+    class="v-form-control" 
+    :value="parentModel" 
+    @keyup="keyupHandler"
+    @focusout="focusoutHandler"
+    @focusin="focusinHandler"
+  >
+  <span class="v-form-control__icon">▾</span>
+  <ul class="suggestions" v-show="showSuggestions" @click="useSuggestion">
+    <li 
+      class="suggestions__item" 
+      v-for="suggestion in suggestions"
+    >
+      {{ suggestion }}
+    </li>
+  </ul>
+</div>
+</template>
+
+
+<script>
+import NetworkService from '../../services/NetworkService';
+
+const SCHOOL_NOT_FOUND_MSG = 'I coudn\'t find my school';
+
+export default {
+  props: {
+    parentModel: String
+  },
+  data() {
+    return {
+      showSuggestions: false,
+      suggestions: []
+    }
+  },
+  methods: {
+    updateParentModel(parent, key, newValue) {
+      parent['user'][key] = newValue;
+    },
+    getSuggestions(query) {
+      return NetworkService.getSuggestions(this, {query})
+        .then((res) => {
+          return Promise.resolve(res.body.suggestions);
+        })
+        .catch((err) => {
+          console.error(err.bodyText);
+          return Promise.resolve([]);
+        });
+    },
+    keyupHandler(e) {
+      let query = e.target.value;
+
+      this.updateParentModel(this.$parent, 'highschool', query);
+
+      this.getSuggestions(query).then((suggestions) => {
+        if (suggestions.length > 0) {
+          this.suggestions = suggestions;
+        }
+        else {
+          this.suggestions = query !== '' ? [SCHOOL_NOT_FOUND_MSG] : [];
+        }
+        this.showSuggestions = this.suggestions.length > 0 ? true : false;
+      }); 
+    },
+    useSuggestion(e) {
+      this.updateParentModel(this.$parent, 'highschool', e.target.innerText);
+      this.showSuggestions = false;
+    },
+    focusoutHandler(e) {
+      setTimeout(() => {
```

Comment:
Because when the user selects a school, both `useSuggestion()` and `focusoutHandler()` get called, and in that case, I need `this.showSuggestions` to be updated a little later to be able to use `e.target.innerText` inside `useSuggestion()`.

Reply:
Add a comment to that effect please! There is probably a way to avoid using the timeout in this scenario but if it's working it's working

Reply:
Added :v: